services/src/messaging/src/main/java/org/openkilda/messaging/command/switches/SwitchDeleteRequest.java

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.messaging.command.switches;
+
+import org.openkilda.messaging.Utils;
+import org.openkilda.messaging.command.CommandData;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Value;
+
+import java.util.Objects;
+
+@Value
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class SwitchDeleteRequest extends CommandData {
+
+    @JsonProperty("switch_id")
+    private String switchId;
+
+    /**
+     * Constructs a delete switch request.
+     *
+     * @param switchId switch id to delete rules from.
+     */
+    @JsonCreator
+    public SwitchDeleteRequest(@JsonProperty("switch_id") String switchId) {
```

Comment:
You can delegate "null check" to the lombok - add `@NonNull` here.

Reply:
Done.